src/static/js/modules/post-filters-form-validation.js

```diff
@@ -31,34 +36,60 @@ function init() {
     * If year or month is missing, the current month or year are added.
     * @param {string} targetDate Date string to format.
     * @param {string} compDate Other date in the date range we're checking.
+    * @param {Event} event Event object used to prevent form submission on error.
     * @returns {string} Formatted targetDate or an empty string.
     * @throws {Error} If the expected YYYY-MM format is wrong.
     */
-    function formatDateString( targetDate, compDate ) {
+    function formatDateString( targetDate, compDate, event ) {
       var formattedDate = targetDate === '' ? compDate : targetDate;
       if ( formattedDate === '' ) return formattedDate;
 
-      if ( /^\d{4}$/.test( formattedDate ) ) {
-        var month = String( new Date().getMonth() + 1 );
-        if ( month.length !== 2 ) month = '0' + month;
-        formattedDate += '-' + month;
-      } else if ( /^\d{2}$/.test( formattedDate ) ) {
-        var year = new Date().getFullYear();
-        formattedDate = year + '-' + formattedDate;
-      } else if ( /^\d{4}-\d{2}$/.test( formattedDate ) === false ) {
-        var msg = 'Unexpected date format! Should be in YYYY-MM format.';
+      var day = '01';
+      var month = String( new Date().getMonth() + 1 );
+      var year = new Date().getFullYear();
+      var dateArray = ( /(?:(\d{1,2})\/)?(?:(\d{1,2})\/)?(\d{2,4})/ )
+                      .exec( formattedDate );
+      var isArray = Array.isArray || function( obj ) {
```

Comment:
This could use the isArray function added in https://github.com/cfpb/cfgov-refresh/pull/618